viewing_party/party.py

```diff
@@ -0,0 +1,255 @@
+    for watched in user_data["watched"]:
+        #for title, movie_title in watched.items():
+            user_movies.add(watched["title"])
+
+    
+    friends_movies = set()
+    for friends_data in user_data["friends"]:
+        for watched in friends_data["watched"]:
+            #for title, movie_titles in watched.items():
+                friends_movies.add(watched["title"])
+
+
+    friends_unique = friends_movies - user_movies
+
+    uni_friends_movies = []
+    for movie in friends_unique:
+        #if movie in friends_unique:
+        uni_friends_movies.append({"title": movie})
+        if len(uni_friends_movies) == 0:
+            return []
+    
+    return uni_friends_movies
+
+#Wave 4
+
+def user_has_not_watched(movie, watched_movies):
+    if movie in watched_movies:
+        return True
+    else:
+        return False
+
+def in_friends_watchlist(movie, watched_by_friends):
+    if movie in watched_by_friends:
+        return True
+    else:
+        return False
+
+def in_user_subscriptions(host, subscriptions):
+    if host in subscriptions:
+        return True
+    else:
+        return False
+
+def get_friends_movies(user_data):
+    titles = set()
+    friends = user_data["friends"]
+    movies = []
+    for friend in friends:
+        for movie in friend["watched"]:
+            if movie["title"] not in titles: 
+                movies.append(movie)
+                titles.add(movie["title"])
+    return movies
+
+def get_available_recs(user_data):
+    friends_movies = get_friends_movies(user_data)
+    movie_recs = []
+
+    if len(friends_movies) == 0:
+        return []
+
+    titles_of_movies_user_watched = set()
+    for movie in user_data["watched"]:
+        titles_of_movies_user_watched.add(movie["title"])
+
+    for movie in friends_movies:
+        if movie["title"] not in titles_of_movies_user_watched:
+            if in_user_subscriptions(movie["host"], user_data["subscriptions"]):
+                movie_recs.append(movie)
+
+    return movie_recs
+
+#wave 5
+def get_new_rec_by_genre(user_data):
+    most_frequent_watched = get_most_watched_genre(user_data)
+    user_not_watched = get_friends_unique_watched(user_data)
+    recommended_movies = []
+
+    #######
+
+#def get_friends_not_watched(user_data):
+    if len(user_data["watched"]) == 0:
+        return []
+    
+    user_movies = []
+    for watched in user_data["watched"]:
+            user_movies.append(watched)
+    
+    friends_movies = []
+    for friends_data in user_data["friends"]:
+        for watched in friends_data["watched"]:
+                friends_movies.append(watched)
+
+    
+    recommend_movies = []
+    most_frequent_watched = get_most_watched_genre(user_data)
+    
+    for movie in friends_movies:
+        if movie not in user_movies and movie["genre"] == most_frequent_watched:
+            if movie in recommend_movies:
+                pass
+            else:
+                recommend_movies.append(movie)
```

Comment:
The nested conditional isn't necessary here, especially since the focus is on appending recommended movies if they're not already in the list. We can add the `movie not in recommend_movies` expression to the outer conditional like so:

```suggestion
        if movie not in user_movies and movie["genre"] == most_frequent_watched and movie not in recommend_movies:            
            recommend_movies.append(movie)
```